test-scripts/resources-tests.sh

```diff
@@ -0,0 +1,551 @@
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to delete property resources, command failed."
+        exit 1
+    fi
+
+    # check that property has been deleted
+    cmd="$ORIGINAL_DIR/bin/${binary} properties get --namespace ecosystemtest \
+    --name $prop_name \
+    --bootstrap $bootstrap \
+    --log -"
+
+    info "Command is: $cmd"
+
+    output_file="$ORIGINAL_DIR/temp/resources-delete-output.txt"
+    $cmd | tee $output_file
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to delete property resource with name used."
+        exit 1
+    fi
+
+    # Check that the previous properties set updated the property value
+    cat $output_file | grep "Total:0" -q
+
+    rc=$?
+    # We expect a return code of 1 because this property should not exist anymore.
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to delete property resource, property remains in namespace."
+        exit 1
+    fi
+
+    rm $output_file
+
+    success "Resources delete with name used seems to have been deleted correctly."
+}
+
+#--------------------------------------------------------------------------
+function properties_resources_delete_invalid_property {
+    h2 "Performing resources properties delete with non-existing property..."
+
+    set -o pipefail # Fail everything if anything in the pipeline fails. Else we are just checking the 'tee' return code.
+
+    input_file="$ORIGINAL_DIR/temp/resources-delete-input.yaml"
+
+    echo """apiVersion: galasa-dev/v1alpha1
+kind: GalasaProperty
+metadata:
+  name: prop.doesnt.exist
+  namespace: ecosystemtest
+data:
+  value: inexistent
+""" > $input_file
+
+    cmd="$ORIGINAL_DIR/bin/${binary} resources delete \
+    --bootstrap $bootstrap \
+    -f $input_file \
+    --log -"
+
+    info "Command is: $cmd"
+
+    $cmd
+    rc=$?
+
+    rm $input_file
+    # We expect a return code of 0 because the api would return an OK status (200) 
+    # as we want this property to not exist
+    if [[ "${rc}" != "0" ]]; then 
+        error "Command should have failed due to non existent property."
+        exit 1
+    fi
+    success "Resources delete with the name of a non existent property correctly throws an error."
+}
+
+#--------------------------------------------------------------------------
+function properties_resources_create_with_name_without_value {
+    h2 "Performing resources create with name parameter, but no value parameter..."
+
+    prop_name="properties.test.name.$PROP_NUM"
+
+    input_file="$ORIGINAL_DIR/temp/resources-create-input.yaml"
+
+    echo """apiVersion: galasa-dev/v1alpha1
+kind: GalasaProperty
+metadata:
+  name: $prop_name
+  namespace: ecosystemtest
+data:
+  value:
+""" > $input_file
+
+    cmd="$ORIGINAL_DIR/bin/${binary} resources create \
+    --bootstrap $bootstrap \
+    -f $input_file \
+    --log -"
+
+    info "Command is: $cmd"
+
+    $cmd
+    rc=$?
+
+    rm $input_file
```

Comment:
personally I like to keep the tests against rc as close to the place where rc is set. The rm of the file could happen after everything ?
I guess you are thinking that if it fails I still want everything cleaned up ? But I think if it fails it's good to have the debris around so you can see the input in more detail. Same pattern throughout this file.

Reply:
Yes, I thought it would be better to clean up as soon as possible, but I will change it to be cleaned at it